modules/ROOT/pages/traversal-framework/traversal_framework_example.adoc

```diff
@@ -0,0 +1,211 @@
+For example, start with this traverser:
+
+[source, java]
+----
+friendsTraversal = db.traversalDescription()
+        .depthFirst()
+        .relationships( Rels.KNOWS )
+        .uniqueness( Uniqueness.RELATIONSHIP_GLOBAL );
+----
+
+
+It should yield the following output (starting from the node with the `name = 'Joe'`):
+
+[source, java]
+----
+(0)
+(0)-[KNOWS,0]->(2)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)<-[KNOWS,4]-(1)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)<-[KNOWS,4]-(1)<-[KNOWS,6]-(5)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)<-[KNOWS,4]-(1)<-[KNOWS,5]-(6)
+----
+
+Then create a new traverser from it, restricting depth to three:
+
+[source, java]
+----
+for ( Path path : friendsTraversal
+        .evaluator( Evaluators.toDepth( 3 ) )
+        .traverse( node ) )
+{
+    output += path + "\n";
+}
+----
+
+This should be the output:
+
+[source, output, role="noheader"]
+----
+(0)
+(0)-[KNOWS,0]->(2)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)
+----
+
+
+In case you want to traverse from depth two to four:
+
+[source, java]
+----
+for ( Path path : friendsTraversal
+        .evaluator( Evaluators.fromDepth( 2 ) )
+        .evaluator( Evaluators.toDepth( 4 ) )
+        .traverse( node ) )
+{
+    output += path + "\n";
+}
+----
+
+This will give the following output:
+
+[source, output, role="noheader"]
+----
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)<-[KNOWS,4]-(1)
+----
+
+To see other useful evaluators, check the link:{org-neo4j-graphdb-traversal-Evaluators}[`org.neo4j.graphdb.traversal.Evaluators`^] Java API or simply implement the link:{org-neo4j-graphdb-traversal-Evaluator}[`org.neo4j.graphdb.traversal.Evaluator`^] interface yourself.
+
+If you are not interested in the Paths, but the Nodes, you can turn the traverser into an iterable of nodes like this:
+
+[source, java]
+----
+for ( Node currentNode : friendsTraversal
+        .traverse( node )
+        .nodes() )
+{
+    output += currentNode.getProperty( "name" ) + "\n";
+}
+----
+
+This will give the following output:
+
+[source, output, role="noheader"]
+----
+Joe
+Sara
+Peter
+Dirk
+Lars
+Lisa
+Ed
+----
+
+You can do this with relationships as well.
+Here is an example:
+
+[source, java]
+----
+for ( Relationship relationship : friendsTraversal
+        .traverse( node )
+        .relationships() )
+{
+    output += relationship.getType().name() + "\n";
+}
+----
+
+
+[source, output, role="noheader"]
+----
+KNOWS
+KNOWS
+KNOWS
+KNOWS
+KNOWS
+KNOWS
+----
+
+== Example
+Below is an example of how to implement a procedure using the Traversal Framework.
+This allows the Traversal Framework to be used side by side with Cypher. The transaction and logger are made available through the Procedure Framework:
```

Comment:
We should provide a link to the pages indicating how to load procedures in the database and an example of how to call `findFriends` from Cypher